mockall/tests/automock_generic_method_with_unknown_size_bounds.rs

```diff
@@ -0,0 +1,38 @@
+// vim: tw=80
+//! generic methods with bounds on their generic parameters
+#![deny(warnings)]
+
+use mockall::*;
+use std::fmt::Debug;
+
+struct X<T: Debug + ?Sized>(Box<T>);
```

Comment:
This test program is overly complicated.  You can eliminate all references to `X` and `Debug` and just test with a method signature like this:
```rust
#[automock]
trait Foo {
    fn foo<T: 'static + ?Sized>(&self, x: Box<T>);
}
```